client/src/components/CreateNewTripForm.jsx

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect } from "react";
+import { fetchParks, createNewTrip, getUserProfileInfo } from "../utils/utils";
+import "../styles/CreateNewTripForm.css";
+import { useNavigate } from "react-router";
+
+const CreateNewTripForm = () => {
+    const [parks, setParks] = useState([]);
+    const [days, setDays] = useState("");
+    const [tripName, setTripName] = useState("");
+    const [parkID, setParkID] = useState("");
+    const nav = useNavigate();
+    const [data, setData] = useState("");
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        await getUserProfileInfo(setData);
+        const body = await createNewTrip(tripName, parkID, data.id);
```

Comment:
have you figured out the error on the first submit?

Reply:
good job! i was suspecting the same issue (some of the values there must be null because new state didn't kick in -- I was guessing if parkID is null) awesome to see you could figure out!